client/app/bundles/course/users/pages/PersonalTimes/index.tsx

```diff
@@ -0,0 +1,79 @@
+import { FC, useEffect, useState } from 'react';
+import { defineMessages, injectIntl, WrappedComponentProps } from 'react-intl';
+import { useDispatch, useSelector } from 'react-redux';
+import { toast } from 'react-toastify';
+import LoadingIndicator from 'lib/components/LoadingIndicator';
+import { AppDispatch, AppState } from 'types/store';
+import PageHeader from 'lib/components/pages/PageHeader';
+import { Paper, Typography } from '@mui/material';
+import { fetchUsers } from '../../operations';
+import {
+  getAllUserMiniEntities,
+  getManageCourseUserPermissions,
+  getManageCourseUsersSharedData,
+} from '../../selectors';
+import UserManagementTabs from '../../components/navigation/UserManagementTabs';
+import SelectCourseUser from '../../components/misc/SelectCourseUser';
+
+type Props = WrappedComponentProps;
+
+const translations = defineMessages({
+  manageUsersHeader: {
+    id: 'course.users.manage.header',
+    defaultMessage: 'Manage Users',
+  },
+  fetchUsersFailure: {
+    id: 'course.users.manage.fetch.failue',
+    defaultMessage: 'Failed to fetch users',
+  },
+  courseUserHeader: {
+    id: 'course.users.personalTimes.header',
+    defaultMessage: 'Course User',
+  },
+});
+
+const PersonalTimes: FC<Props> = (props) => {
+  const { intl } = props;
+  const [isLoading, setIsLoading] = useState(true);
+  const users = useSelector((state: AppState) => getAllUserMiniEntities(state));
+  const permissions = useSelector((state: AppState) =>
+    getManageCourseUserPermissions(state),
+  );
+  const sharedData = useSelector((state: AppState) =>
+    getManageCourseUsersSharedData(state),
+  );
+  const dispatch = useDispatch<AppDispatch>();
+
+  useEffect(() => {
+    dispatch(fetchUsers(false))
```

Comment:
I am not really in favour of fetching everything. We only need the id and name right?

Reply:
Same for the other instance of fetchUsers